### PR TITLE
Issue 6

### LINKED ISSUE #6: Create A New Note
At the bottom of the `reminders.hbs` template, there should be a button that routes the user to `/reminders/new` where users can create a new reminder.

* The new reminder should have inputs for the title, date, and notes.
* The new reminder should have a button for submitting the new note.
* Pressing the button should submit a new reminder to the database and immediately displays in the reminders list.
**Hot Tip**: There are no tests set up for this user story. So, you’ll have to write an acceptance test.

### PR DESCRIPTION
**Pull request by jksmall0631**
## Purpose

At the bottom of the reminders.hbs template, there should be a button that routes the user to /reminders/new where users can create a new reminder.

The new reminder should have inputs for the title, date, and notes.
The new reminder should have a button for submitting the new note.
Pressing the button should submit a new reminder to the database and immediately displays in the reminders list.
Hot Tip: There are no tests set up for this user story. So, you’ll have to write an acceptance test.

[Issue #6](https://github.com/turingschool-projects/1610-remember-1/issues/6)

## Approach

We added  a 'new' route as well as a 'new' button to the reminders list template. We then built out our 'new' template to include text fields and a submit button. We added an action to the 'new' form that would submit the desired text to the store and render the new title on the screen. Followed up by writing both acceptance and integration tests.

### Learning

[Ember Docs Nested Routes](https://guides.emberjs.com/v2.7.0/tutorial/subroutes/)
[Input fields on stack overflow](http://stackoverflow.com/questions/14450862/ember-js-input-fields)
[Ember Docs Actions](https://guides.emberjs.com/v2.3.0/templates/actions/)

### Open Questions and Pre-Merge TODOs

- [x] add new nested route
- [x] add new reminder button
- [x] add input fields and submit button
- [x] create a template for new reminder functionality
- [x] add more acceptance tests
- [x] add integration tests

### Test coverage 

We added two more acceptance tests and a bunch of integration tests on our new route component. We tested that the page would go where we wanted it to go and listened to our click, and that it rendered everything we were expecting it to.

### Follow-up tasks

- [Issue #7](https://github.com/turingschool-projects/1610-remember-1/issues/7)

#### Before
screen shot 2017-02-09 at 5 46 08 pm

#### After
screen shot 2017-02-11 at 7 23 04 pm
screen shot 2017-02-11 at 7 23 15 pm
screen shot 2017-02-11 at 8 09 11 pm
screen shot 2017-02-11 at 8 30 01 pm

close #6

### CODE REVIEW

File: app/templates/components/new-reminder.hbs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  </label>
+  <label>
+    Date
+    {{input class="spec-input-date" value=date}}
+  </label>
+  <label>
+    Notes
+    {{textarea class="spec-textarea-notes" value=notes}}
+  </label>
+  <button class='spec-reminder-add--submit this-shit-is-bananas-B-A-N-A-N-A-S'>Submit</button>
```
Comment (martensonbj) [inline app/templates/components/new-reminder.hbs]: ?? Tell me more...

File: app/templates/components/reminders-list.hbs [hunk 2]
```diff
@@ -1,5 +1,11 @@
 <h1>{{title}}</h1>
```
Comment (martensonbj) [inline app/templates/components/reminders-list.hbs]: Where is this title coming from?

**Review by martensonbj — CHANGES_REQUESTED** [review]
Couple clarifying questions

File: app/templates/components/new-reminder.hbs (hunk 1 above)
Comment (jksmall0631) [reply app/templates/components/new-reminder.hbs]: It was a late night last night, Brenna.

File: app/templates/components/reminders-list.hbs (hunk 2 above)
Comment (jksmall0631) [reply app/templates/components/reminders-list.hbs]: That's coming from reminders.hbs. Definitely not necessary but just wanted to try to pass it through...

**Comment by jksmall0631** [comment]
@martensonbj We just made those changes, check it out!

**Comment by Tman22** [comment]
The bananas have been moved to a suitable area..

**Pull request merged**